pkg/store/consul/consultest/fake_preparer.go

```diff
@@ -66,17 +62,33 @@ func (f *FakePreparer) Enable() {
 					if err != nil {
 						f.logger.Errorf("Error setting pod: %v", err)
 					}
-					delete(podsToDelete, manifestResult)
 				}
+
+				allReality, _, err := f.podStore.AllPods(consul.REALITY_TREE)
+				if err != nil {
+					f.logger.Errorf("error getting all reality pods: %v", err)
+					continue
+				}
+
 				// Delete pods that are missing from intent
-				for manifestResult := range podsToDelete {
-					_, err = f.podStore.DeletePod(
-						consul.REALITY_TREE,
-						manifestResult.PodLocation.Node,
-						manifestResult.PodLocation.PodID,
-					)
-					if err != nil {
-						f.logger.Errorf("Error deleting pod: %v", err)
+				for _, manifestResult := range allReality {
+					found := false
+					for _, intentPod := range allPods {
+						if intentPod.PodLocation.Node == manifestResult.PodLocation.Node {
```

Comment:
why does it only check the `Node` and not the `PodID`?

This means that if I have intent: `node1/b`, and reality: `node1/a, node1/b`, then `node1/a` doesn't get deleted when it should, right?